src/ui/pages/AssetInfo/AssetResources.tsx

```diff
@@ -0,0 +1,365 @@
+import { UnstyledButton } from 'src/ui/ui-kit/UnstyledButton';
+import { VStack } from 'src/ui/ui-kit/VStack';
+import DownIcon from 'jsx:src/ui/assets/chevron-down.svg';
+import CopyIcon from 'jsx:src/ui/assets/copy.svg';
+import CheckIcon from 'jsx:src/ui/assets/check_double.svg';
+import LinkIcon from 'jsx:src/ui/assets/new-window.svg';
+import ArrowLeftIcon from 'jsx:src/ui/assets/arrow-left.svg';
+import XIcon from 'jsx:src/ui/assets/x-logo.svg';
+import WarpcastIcon from 'jsx:src/ui/assets/warpcast-logo.svg';
+import DexscreenerIcon from 'jsx:src/ui/assets/dexscreener-logo.svg';
+
+function MainImplementationButton({
+  implementation,
+}: {
+  implementation: { address: string | null; network: NetworkConfig };
+}) {
+  const { address, network } = implementation;
+  const { handleCopy, isSuccess } = useCopyToClipboard({
+    text: address || 'null',
+  });
+
+  if (!address) {
+    return (
+      <Button
+        as={UnstyledAnchor}
+        rel="noopener noreferrer"
+        target="_blank"
+        href={network.explorer_home_url || ''}
+        aria-label="Copy token address"
+        size={36}
+        kind="neutral"
+        className="parent-hover"
+        style={{
+          ['--parent-content-color' as string]: 'var(--neutral-400)',
+          ['--parent-hovered-content-color' as string]: 'var(--neutral-700)',
+          ['--button-background' as string]: 'var(--neutral-200)',
+          ['--button-background-hover' as string]: 'var(--neutral-300)',
+          paddingInline: 8,
+        }}
+      >
+        <HStack gap={4} alignItems="center">
+          <HStack gap={6} alignItems="center">
+            <img
+              src={network.icon_url}
+              alt={network.name}
+              width={24}
+              height={24}
+            />
+            <UIText kind="small/accent">Explorer</UIText>
+          </HStack>
+          <LinkIcon
+            className="content-hover"
+            style={{ width: 20, height: 20 }}
+          />
+        </HStack>
+      </Button>
+    );
+  }
+  return (
+    <Button
+      onClick={handleCopy}
+      aria-label="Copy token address"
+      size={36}
+      kind="neutral"
+      className="parent-hover"
+      style={{
+        paddingInline: 8,
+        ['--parent-content-color' as string]: 'var(--neutral-400)',
+        ['--parent-hovered-content-color' as string]: 'var(--neutral-700)',
+        ['--button-background' as string]: 'var(--neutral-200)',
+        ['--button-background-hover' as string]: 'var(--neutral-300)',
+      }}
+    >
+      <HStack gap={4} alignItems="center">
+        <HStack gap={6} alignItems="center">
+          <img
+            src={network.icon_url}
+            alt={network.name}
+            width={24}
+            height={24}
+          />
+          <UIText kind="small/accent">
+            {middleTruncate({
+              value: address,
+              leadingLettersCount: 5,
+              trailingLettersCount: 4,
+            })}
```

Comment:
Is this better for some reason than `truncateAddress`?

Reply:
I need to check. Tbh, I just forgot about other ways to truncate addresses